src/trove/string-dict.scrbl

```diff
@@ -225,6 +236,102 @@ check:
 end
 }
 
+@sd-method["merge"
+  #:contract (a-arrow (SD-of "a") (SD-of "a") (SD-of "a"))
+  #:args (list (list "self" #f) (list "other" #f))
+  #:return (SD-of "a")
+]
+
+Returns a new immutable string-dict that has the keys of both
+the original string-dict and the @pyret{other} string-dict. Choose a key's
+value to be the one it has in the @pyret{other} string-dict, if it has one;
+otherwise it has the value from the original string-dict.
+
+@examples{
+check:
+  sd1 = [string-dict: "a", 5, "c", 4]
+  sd2 = [string-dict: "a", 10, "b", 6]
+  sd3 = sd1.merge(sd2)
+  sd3 is [string-dict: "a", 10, "b", 6, "c", 4]
+  sd4 = sd2.merge(sd1)
+  sd4 is [string-dict: "a", 5, "b", 6, "c", 4]
+  sd2.merge(sd1).merge(sd1) is sd2.merge(sd1)
+end
+}
+
+@sd-method["keys-list"
+  #:contract (a-arrow (SD-of "a") (L-of S))
+  #:args (list (list "self" #f))
+  #:return (L-of S)
+]
+
+Returns the list of keys in the immutable string-dict, in some order.
+
+@examples{
+check:
+  sd2 = [string-dict: "a", 10, "b", 6]
+  sd2.keys-list() is [list: "a", "b"]
+end
+}
+
+@sd-method["map-keys"
+  #:contract (a-arrow (SD-of "a") (a-arrow "a" "b") (L-of "b"))
+  #:args (list (list "self" #f) (list "f" #f))
+  #:return (L-of "b")
+]
+
+Applies the function @pyret{f} to each key in the immutable string-dict, and
+returns a list of the returned values in some order.
+
+@examples{
+fun one-of(ans, elts):
+  is-some(for find(elt from elts):
+    ans == elt
+  end)
+end
+
+check:
+  sd2 = [string-dict: "a", 10, "b", 6]
+  sd2.map-keys(lam(x): string-append(x, x) end) is%(one-of)
+    [list: [list: "aa", "bb"], [list: "bb", "aa"]]
+end
+}
+
+@sd-method["fold-keys"
+  #:contract (a-arrow (SD-of "a") (a-arrow S "b") "b")
+  #:args (list (list "self" #f) (list "f" #f) (list "init" #f))
+  #:return "b"
+]
+
+Returns the result of folding the function @pyret{f} across the keys in the
+immutable string-dict, in some order, starting with @pyret{init} as the base value.
+
+@examples{
+check:
+  sd2 = [string-dict: "a", 10, "b", 6]
+  sd2.fold-keys(lam(x, acc): acc + string-length(x) end, 0) is 2
+end
+}
+
+@sd-method["each-key"
+  #:contract (a-arrow (SD-of "a") (a-arrow S No) No)
+  #:args (list (list "self" #f) (list "f" #f))
+  #:return No
+]
+
+Calls the function @pyret{f} on each key in the immutable string-dict,
```

Comment:
The point of this is the side effects of `f`, right?  Probably worth mentioning.